### PR TITLE
Block validation: Ensure transactions are sorted

### PR DESCRIPTION
**Pull request by Geod24**
So that we get a more stable block creation *and* can leverage this for block propagation.

### CODE REVIEW

**Comment by AndrejMitrovic** [comment]
LGTM. The docker failure is probably because the `getChainedTransaction` function is duplicated for the integration tests. So there are missing `sort` calls somewhere in those tests.

**Comment by AndrejMitrovic** [comment]
> LGTM. The docker failure is probably because the getChainedTransaction function is duplicated for the integration tests. So there are missing sort calls somewhere in those tests.

Well I expect this to become an error later, but for now it seems something else is wrong https://circleci.com/gh/bpfkorea/agora/29?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Comment by Geod24** [comment]
Slapped the CI into submission.

File: source/agora/consensus/data/Block.d [hunk 1]
```diff
@@ -374,6 +374,7 @@ public Block makeNewBlock (const ref Block prev_block, Transaction[] txs) @safe
     block.header.prev_block = prev_block.header.hashFull();
     block.header.height = prev_block.header.height + 1;
     block.txs ~= txs;
+    block.txs.sort!((a, b) => hashFull(a) < hashFull(b));
```
Comment (Geod24) [inline source/agora/consensus/data/Block.d]: I think I can remove the predicate now

Comment (Geod24) [reply source/agora/consensus/data/Block.d]: Done

**Comment by Geod24** [comment]
Green!

**Review by AndrejMitrovic — APPROVED** [review]

**Pull request merged**